src/libostree/ostree-sysroot.c

```diff
@@ -815,6 +815,24 @@ list_deployments_process_one_boot_entry (OstreeSysroot               *self,
     return FALSE;
 
   ostree_deployment_set_bootconfig (deployment, config);
+  char **overlay_initrds = ostree_bootconfig_parser_get_overlay_initrds (config);
+  g_autoptr(GPtrArray) initrds_chksums = NULL;
+  for (char **it = overlay_initrds; it && *it; it++)
+    {
+      const char *basename = glnx_basename (*it);
+      if (strlen (basename) != (_OSTREE_SHA256_STRING_LEN + strlen (".img")))
```

Comment:
It'd be nice to allow people to have a "short name" for their overlay initrd to describe its purpose without needing to unpack it to look.

Basically we append the checksum to the name they provide?

Reply:
IOW it's `/boot/ostree/initramfs-overlays/kdump-41af286dc0b172ed2f1ca934fd2278de4a1192302ffa07087cea2682e7d372e3.img` and not just `/boot/ostree/initramfs-overlays/41af286dc0b172ed2f1ca934fd2278de4a1192302ffa07087cea2682e7d372e3.img`

Reply:
If you prefer we can do this as a followup too.

Reply:
Are you suggesting expanding the overlay initrd staging API as e.g.:

```diff
diff --git a/src/libostree/ostree-sysroot.h b/src/libostree/ostree-sysroot.h
index 3a3b6a77..ef5a353c 100644
--- a/src/libostree/ostree-sysroot.h
+++ b/src/libostree/ostree-sysroot.h
@@ -189,6 +189,7 @@ gboolean ostree_sysroot_write_deployments_with_options (OstreeSysroot     *self,
 _OSTREE_PUBLIC
 gboolean ostree_sysroot_stage_overlay_initrd (OstreeSysroot  *self,
                                               int             fd,
+                                              const char      short_id,
                                               char          **out_checksum,
                                               GCancellable   *cancellable,
                                               GError        **error);
```

?

Hmm, or maybe we could change the semantics of `OstreeSysrootDeployTreeOpts.overlay_initrds` to have the checksum be optionally prefixed by an ID?

:thinking: Honestly, while the code here does support multiple overlays, I think in practice it's not something that will be used often. E.g. in the rpm-ostree use case, there is just a single "initramfs-etc" initrd, and my hope is that it's the one way through which users interact with this feature.

Reply:
I think I'd vote for leaving the API simple and we can do the second approach mentioned above if it comes up? (Because the first approach would obviously be an API break.)

Reply:
OK, fine by me!